File: adventofcode/2020/15/first.js
const { read, range } = require("../../../util");

const input = read("./data.txt");

const findLastIndexes = (series, number) => {
  const res = [];
  for (let i = series.length - 1; i >= 0; i--) {
    if (series[i] === number) {
      res.push(i + 1);
    }

    if (res.length >= 2) {
      break;
    }
  }

  return res;
};

input
  .then((content) => {
    const series = content[0].split(",").map(Number);

    for (let i = series.length; i < 30000000; i++) {
      const lastNumber = series[i - 1];
      //   console.log(series, lastNumber, i);

      const lastIndexes = findLastIndexes(series, lastNumber);
      //   console.log(lastIndexes);
      if (lastIndexes.length < 2) {
        series.push(0);
      } else {
        series.push(lastIndexes[0] - lastIndexes[1]);
      }
    }

    return series[series.length - 1];
  })
  .then(console.log);

File: adventofcode/2021/6/first.py
f = open("input.txt", "r")
input = [int(val) for val in f.read().split(',')]

def sm(state):
    new_state = []
    for val in state:
        if val == 0:
            new_state.append(6)
            new_state.append(8)
        else:
            new_state.append(val - 1)
    
    return new_state

def recurse_sm(state, cycle):
    return recurse_sm(sm(state), cycle - 1) if cycle != 0 else state

def main(series):
    cycles = 80

    return len(recurse_sm(series, cycles))

print(main(input))
